### PR TITLE
CI: add GitHub Actions for GreenplumPython with postgres.

### PR DESCRIPTION
**Pull request by yihong0618**
Also fix all test.

### CODE REVIEW

File: tests/test_db.py [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+    db = gp.database(
+        host="localhost",
+        dbname="gpadmin",
+        user=environ.get("POSTGRES_USER"),
+        password=environ.get("POSTGRES_PASSWORD"),
+    )
     result = db.execute("SELECT version()")
     for row in result:
-        assert "Greenplum" in row["version"]
+        assert "Greenplum" in row["version"] or row["version"].startswith("PostgreSQL")
```
Comment (yihong0618) [inline tests/test_db.py:15]: Postgres version startswith `PostgreSQL`

File: tox.ini [hunk 2]
```diff
@@ -15,7 +15,10 @@ deps =
 deps = {[base]deps}
 commands =
     pytest {posargs}
-passenv=PGPORT
+passenv=
+        PGPORT
+        POSTGRES_PASSWORD
+        POSTGRES_USER
```
Comment (yihong0618) [inline tox.ini:21]: more env to export add here.

**Review by Quasimewdo — COMMENTED** [review]
LGTM

**Comment by beeender** [comment]
image

The concourse CI is missing. The web hook for concourse PR resource has a known issue with PRs from forked repo. You probably need to click the `refresh` button to trigger it manually.

**Comment by yihong0618** [comment]
> image
> 
> The concourse CI is missing. The web hook for concourse PR resource has a known issue with PRs from forked repo. You probably need to click the `refresh` button to trigger it manually.

Done

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        cache: pip
+    - name: Install Dependencies
+      run: |
+        pip install tox==3.25.0
+    - name: Create plpython3u language and environment
+      run: |
+        psql postgres://postgres:postgres@localhost/gpadmin -c "CREATE LANGUAGE plpython3u;"
+    - name: Run Tests
+      run: |
+        export POSTGRES_PASSWORD=postgres
```
Comment (beeender) [inline .github/workflows/test.yml:53]: I think if you use  `PGPASSWORD` and `PGUSER`, you don't even need to change the python files.

Comment (yihong0618) [reply .github/workflows/test.yml:53]:
```python
    db = gp.database(
        host="localhost",
        dbname="gpadmin",
        user=environ.get("POSTGRES_USER"),
        password=environ.get("POSTGRES_PASSWORD"),
```
can the user and password just pass?

Comment (yihong0618) [reply .github/workflows/test.yml:53]: let me try.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    services:
+      postgres:
+        # we need postgres with plpython extension postgres base docker hub image did not have that
+        # I use this one
+        image: thorinschiffer/postgres-plpython:${{ matrix.postgres-version }}
+        env:
+          POSTGRES_PASSWORD: postgres
+          # GreenplumPython use gpadmin as db name
+          POSTGRES_DB: gpadmin
```
Comment (fanfuxiaoran) [inline .github/workflows/test.yml:34]: "gpadmin" as a test database name is a little strange,
```suggestion
          POSTGRES_DB: greenplum_python_test
```

File: tests/test_unaryExpr.py [hunk 5]
```diff
@@ -1,11 +1,18 @@
...
 def db():
-    db = gp.database(host="localhost", dbname="gpadmin")
+    db = gp.database(
+        host="localhost",
+        dbname="gpadmin",
+        user=environ.get("POSTGRES_USER"),
+        password=environ.get("POSTGRES_PASSWORD"),
+    )
     yield db
     db.close()
```
Comment (fanfuxiaoran) [inline tests/test_unaryExpr.py:17]: Can we extract this function into a common python file? as it appears in many test files.

Comment (yihong0618) [reply tests/test_unaryExpr.py:17]: I think so.

Comment (yihong0618) [reply tests/test_unaryExpr.py:17]: will try

File: tests/test_binaryExpr.py [hunk 6]
```diff
@@ -1,11 +1,18 @@
+from os import environ
+
 import pytest
 
 import greenplumpython as gp
 
 
-@pytest.fixture
+@pytest.fixture()
```
[the remark this answers is not in the document]
Comment (yihong0618) [reply tests/test_binaryExpr.py]: after some search we do not need that `session`

File: .github/workflows/test.yml (hunk 3 above)
Comment (fanfuxiaoran) [reply .github/workflows/test.yml]: Agree to use PGUSER and PGPASSWORD

Comment (yihong0618) [reply .github/workflows/test.yml]: database name PGDB?

**Review by fanfuxiaoran — APPROVED** [review]

**Review by Quasimewdo — APPROVED** [review]

**Pull request merged**